Source/JavaScriptCore/wasm/WasmIPIntGenerator.cpp

```diff
@@ -986,21 +988,46 @@ PartialResult WARN_UNUSED_RETURN IPIntGenerator::addBlock(BlockSignature signatu
     return { };
 }
 
-PartialResult WARN_UNUSED_RETURN IPIntGenerator::addLoop(BlockSignature signature, Stack& oldStack, ControlType& block, Stack& newStack, uint32_t)
+PartialResult WARN_UNUSED_RETURN IPIntGenerator::addLoop(BlockSignature signature, Stack& oldStack, ControlType& block, Stack& newStack, uint32_t loopIndex)
 {
     splitStack(signature, oldStack, newStack);
     block = ControlType(signature, BlockType::Loop);
     block.m_pendingOffset = -1; // no need to update!
 
     // Allocate space in metadata
-    m_metadata->m_repeatedControlFlowInstructionMetadataOffsets.append(m_metadata->m_metadata.size());
-    m_metadata->addBlankSpace(8);
-    condenseControlFlowInstructions();
-
+    auto size = m_metadata->m_metadata.size();
+    m_metadata->addBlankSpace(1);
+    WRITE_TO_METADATA(m_metadata->m_metadata.data() + size, getCurrentInstructionLength(), uint8_t);
     // No -1 because we can just have it directly go to the instruction after
     // No point running `loop` since in IPInt it's just a nop
     block.m_pc = m_parser->offset() - m_metadata->m_bytecodeOffset;
     block.m_mc = m_metadata->m_metadata.size();
+
+    // Loop OSR
+
+    // We don't care what we're putting in as long as we put in something to represent a value
+    // Will get everything off the stack
+    Vector<VirtualRegister> osrEntryData;
```

Comment:
Can we just initialize this vector directly to be the right size?

Reply:
Updated to precalculate the size, and then initialize the vector directly with that size.